enhancements/testing/openshift-tests-extension.md

```diff
@@ -0,0 +1,722 @@
+```
+
+The "list" verb will not be provided a kubeconfig and must output tests based solely
+on environment information. If no environment arguments are provided, all tests must be listed.
+
+Listings are formatted as JSONL with one test description per line. A full listing contains
+zero or more test description objects. The following example shows an abbreviated JSONL listing 
+for three tests.
+```python
+{ "name": "T1", }
+{ "name": "T2", }
+{ "name": "T3", }
+```
+
+A test description object contains a number of attributes to inform `openshift-tests`
+discovery and planning. Note that the following annotated example would be encoded into a single 
+line of output in actual listing output.
+```python
+{
+  
+  # Base, human-readable test name. 
+  "name": "openssl version compliance",
+
+  # If a test name is updated at any time in the future,
+  # originalName must report the original name of the 
+  # testing logic. This allows component readiness
+  # to display the human-readable version of the test
+  # name while considering test runs across name changes.
+  "originalName": "security version compliance",
+    
+  # Optional subcomponent information.
+  "subcomponent": "ui",
+    
+  # Labels are text strings will can be used like 
+  # Ginkgo labels to group and run tests. Until labels
+  # are part of bigquery data, labels will be appended
+  # to test names with "[label]" so that name based
+  # suite construction will work. 
+  "labels": [
+    "sig-..."
+  ],
+  
+  # Tags are key=value pairs that can be used to 
+  # further classify tests.
+  "tags": {
+    "key": "value"
+  },
+  
+  "resources": {
+    # Tests can request to be run in isolation
+    # from others at different levels:
+    # "instance" - avoids being called along with a conflicting
+    #              test in a single run-test invocation.
+    # "exec"     - avoids being called at the same time in a
+    #              across all parallel extension binary invocations.
+    # "bucket"   - avoids being called in the same planning bucket.
+    # If two tests have a conflict name in common, the isolation
+    # mode will apply. "*" is a special conflict name will 
+    # ensures complete isolation in the requested mode (e.g.
+    # guaranteeing the test will be the only one to
+    # a given invocation of run-test if mode=instance).
+    "isolation": {
+      "mode": "exec",
+      "conflict":  [
+        "gpu",
+      ]
+    },
+      
+    # Tests that require significant resources can identify
+    # that fact so that the planning algorithm will not
+    # overdraw on allocatable pod resources for a set of
+    # tests run in parallel.
+    "memory": "1Gi",
+    
+    # If a duration can be roughly predicted, inclusion of this
+    # information may improve the test execution 
+    # planning algorithm performance.  
+    "duration": "2s",
+      
+    # Timeout information may also support efficient
+    # planning.
+    "timeout": "16s",

```

Comment:
I'm not sure how to implement basically everything here other than labels.  Ginkgo doesn't provide any facility to add metadata to tests other than labels.

We could write some helpers that apply metadata in a particular way, but it'd probably have to all be stuffed into labels - unless you have another idea?

Reply:
TBD but this can be resolved.  We can handle it by walking the specs in injecting the data, or labeling the tests.